HHS/Head-Start-TTADP
# [TTAHUB-3281] manual mark goals similar

## Description of change

Implemented the ability to manually mark goals as similar within the Head Start TTA Hub. This new feature allows users to select multiple goals and group them together if they are determined to be similar. The change includes updates to the frontend UI to support this functionality, new routes, and backend modifications to process and store the similarity groupings.


## How to test

1. Navigate to the goals management section of the application.
2. Select multiple goals to mark them as similar.
3. Verify that the selected goals are grouped correctly.
4. Ensure the backend processes and stores these groupings as expected.
5. Run the updated unit tests to confirm all are passing.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3281


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
